### PR TITLE
feat: consistent paths in messages

### PR DESCRIPTION
**Pull request by P0lip**
Addresses https://github.com/stoplightio/spectral/issues/815

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/__tests__/linter.test.ts [hunk 1]
```diff
@@ -642,7 +642,7 @@ responses:: !!foo
       }),
       expect.objectContaining({
         code: 'oas3-valid-schema-example',
-        message: '"foo.example" property type should be number',
+        message: 'property type should be number',
```
Comment (P0lip) [inline src/__tests__/linter.test.ts]: @philsturgeon is this something you expected to see?
I believe there is no point in printing paths when it exists in the document.
The only reasonable case is to print them when the property **does not** exist in the document and you want to indicate that, i.e. "XYZ property is not truthy".

We could keep it configurable though and let users decide, but leaving the call up to you all.

cc @marbemac

Comment (philsturgeon) [reply src/__tests__/linter.test.ts]: I would expect to see "example property type should be number" or `"The property type of \"example\" should be number."`

Comment (marbemac) [reply src/__tests__/linter.test.ts]: I agree that having just that little bit more context in the error message is helpful. Suggest we keep it in the pretty message.

**Comment by P0lip** [comment]
@marbemac @philsturgeon 
the last remaining question is whether we are making paths configurable - right now we always try to display the property name, but some want to have the whole path printed.
What do you think?

**Comment by philsturgeon** [comment]
If I had to guess at the use case for people who want the whole path, they are probably using it programmatically do if we make it available in the object (and it can be inserted into the message string via a {{path}} variable) then both JS and CLI users should be happy. Then we can keep it property whenever possible by default.

**Comment by marbemac** [comment]
> whether we are making paths configurable

Suggest we punt this to later - not convinced we need it so might as well save the time now :).

**Comment by P0lip** [comment]
> If I had to guess at the use case for people who want the whole path, they are probably using it programmatically do if we make it available in the object (and it can be inserted into the message string via a {{path}} variable) then both JS and CLI users should be happy. Then we can keep it property whenever possible by default.

We do expose `{{path}}` now, so your use case would be covered :)

**Pull request merged**